File: 594. Longest Harmonious Subsequence/findLHS.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findLHS = function (nums) {
    let mp = new Map()
    let res = 0

    for (let num of nums) {
        if (!mp.has(num)) {
            mp.set(num, 0)
        }
        mp.set(num, mp.get(num) + 1)
    }

    for (let [num, cnt] of mp) {
        if (mp.has(num + 1)) {
            let sum = cnt + mp.get(num + 1)
            res = Math.max(res, sum)
        }
    }

    return res
};

File: 594. Longest Harmonious Subsequence/findLHS.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <unordered_map>

using namespace std;

class Solution {
    public:
        int findLHS(vector<int>& nums) {
            unordered_map<int,int> mp;
            int res = 0;
    
            for(auto num : nums){
                mp[num]+=1;
            }
    
            for(const auto& [num, cnt] : mp){
                if(mp.count(num+1)){
                    int sum = cnt + mp[num+1];
                    res = max(res, sum);
                }
            }
    
            return res;
        }
    };